app/Main.hs

```diff
@@ -22,10 +23,14 @@ main = do
 
     runMigration migrateAll
 
-    -- prepare users, insert users, get entities. reminder:
-    -- Section { sectionUsername :: [String], sectionParagraphs :: [[String]] }
-    let users = map (\section -> User $ sectionUsername section) sections :: [User]
-    _ <- insertMany users
+    -- insert user and pagaraph in one go
+    forM_ sections $ \section -> do
+      let user = User $ sectionUsername section
+      userId <- insert user
+      forM_ (sectionParagraphs section) $ (\p -> do
```

Comment:
Eh, I don't have a strong opinion, but I'd pick one or the other. Either just parens or just `$`. Above you're just using `$` with no parens, and I think it'll read fine. You could even do a transformation from `map` (which won't compile) to `mapM` to `mapM_` and finally to `forM_` if you wanted to show that this thing isn't magic.